book/src/queries/schema_agreement.md

```diff
@@ -0,0 +1,90 @@
+# Schema agreement
+
+Sometimes after performing queries some nodes have not been updated so we need a mechanism that checks if every node have agreed schema version.
+There are four methods in `Session` that assist us. 
+Every method raise `QueryError` if something goes wrong, but they should never raise any errors, unless there is a DB or connection malfunction.
+
+### Checking schema version
+`Session::fetch_schema_version` returns an `Uuid` of local node's schema version. 
+
+```rust
+# extern crate scylla;
+# use scylla::Session;
+# use std::error::Error;
+# async fn check_only_compiles(session: &Session) -> Result<(), Box<dyn Error>> {
+println!("Local schema version is: {}", session.fetch_schema_version().await?);
+# Ok(())
+# }
+```
+
+### Awaiting schema agreement
+
+`Session::await_schema_agreement` returns a `Future` that can be `await`ed on as long as schema is not in an agreement.
+
+```rust
+# extern crate scylla;
+# use scylla::Session;
+# use std::error::Error;
+# async fn check_only_compiles(session: &Session) -> Result<(), Box<dyn Error>> {
+session.await_schema_agreement().await?;
+# Ok(())
+# }
+```
+
+### Awaiting with timeout
+We can also set timeout in miliseconds with `Session::await_timed_schema_agreement`.
+It takes one argument, an `std::time::Duration` value that tells how long in milliseconds our driver should await for schema agreement. If the timeout is met the return value is `false` otherwise it is `true`.
```

Comment:
The "in milliseconds" phrase is not needed anymore.

Reply:
done